canonical/loki-k8s-operator
# new loki_push_api lib version

## Issue


This PR fixes #307

## Solution


A new version of loki_push_api lib

## Context


Before this PR, it was not possible to send logs from more than one container of the charm.


## Testing Instructions


1. Pack the `log-proxy-tester`
2. Deploy this bundle:
```yaml
bundle: kubernetes
applications:
  loki:
    charm: loki-k8s
    channel: edge
    revision: 101
    resources:
      loki-image: 84
    scale: 1
    constraints: arch=amd64
    storage:
      active-index-directory: kubernetes,1,1024M
      loki-chunks: kubernetes,1,1024M
    trust: true
  tester:
    charm: ./tests/integration/log-proxy-tester/log-proxy-tester_ubuntu-20.04-amd64.charm
    scale: 1
    constraints: arch=amd64
    resources:
      workload-image: ghcr.io/sed-i/flog:latest
relations:
- - tester:log-proxy
  - loki:logging
```
3. Check that there are 2 series in Loki:
```shell
$ curl -s http://10.1.38.92:3100/loki/api/v1/series | jq
{
  "status": "success",
  "data": [
    {
      "job": "juju_logproxy_1793d63b_tester",
      "filename": "/bin/driver.log",
      "container": "workload-b",
      "juju_unit": "tester/0",
      "juju_model_uuid": "1793d63b-402f-475a-8b6c-96853029fdff",
      "juju_model": "logproxy",
      "juju_charm": "log-proxy-tester",
      "juju_application": "tester"
    },
    {
      "juju_charm": "log-proxy-tester",
      "juju_application": "tester",
      "job": "juju_logproxy_1793d63b_tester",
      "filename": "/bin/driver.log",
      "container": "workload-a",
      "juju_unit": "tester/0",
      "juju_model_uuid": "1793d63b-402f-475a-8b6c-96853029fdff",
      "juju_model": "logproxy"
    }
  ]
}
```
